rfc/2023/2023-05-01-va-rfc-frontend-performance.md

```diff
@@ -0,0 +1,156 @@
+
+## Risks
+
+
+>List the risks of this approach
+> * There are always risks. What are the risks of this solution?
+> * These are the things people will bring up in opposition to your idea or plans. Acknowledge them.
+
+- There are large assets on the page such as Google Analytics and Digital Voice scripts that we cannot optimize or remove. Performance budgets can and should acknowledge this.
+- Removing duplicate scripts can create dependencies between bundles. Given our Webpack configuration already injects "shared modules", this dependency is already expected, we're just missing the de-duplication step.
+- Applications may require lots of refactoring to reach a performance budget goal, or teams may simply not have a lot of time to work on performance. Budgets can be adopted gradually, and process created for customizing budgets and allowing exceptions when there are significant barriers to increasing performance.
+- Teams may need help improving performance. We can adopt a "we're all in this together" approach and empower teams to help each other
+- This could be perceived as a Platform responsibility. If VFS teams are empowered to make or propose improvements, this work can and should be shared across VFS and Platform teams.
+- People may question the data informing this proposal. The VA-specific research mentioned is from 2019, and broadband adoption has likely changed since then. The guidance provided by "The Performance Inequality Gap" is based on global measures, not US or Veteran-specific measures. We could spend time gathering more data on the performance experiences of VA.gov users.
+
+## Alternatives
+
+> List the alternative approaches
+> * There are always alternatives. What other alternative solutions were considered? Not considered?
+> * What are their strengths, weaknesses, risks? Why weren’t they chosen?
+> * Do not allow bias of a solution to show in this section, ensure each alternative has been considering seriously or do not list it in this section
+
+One alternative would be to adopt an hypermedia-centric approach and have `vets-api` generate HTML documents and "partials" in coordination with a lightweight hypertext-centric framework such as [htmx](https://htmx.org), a ~14KiB JavaScript library. This would shift a lot of logic to the server and minimize the amount of JavaScript, the major contributor to the performance issues facing VA.gov. This would represent a paradigm shift in how VA.gov applications are built, and require a significant investment in time and resources to adopt across all applications. This shift could be done incrementally, but still represents a major architectural change.
```

Comment:
link to the other RFC that discusses alternate frameworks. that other RFC could help teams achieve their performance budgets as discusses in this RFC